CMakeLists.txt

```diff
@@ -1480,14 +1480,14 @@ IF(CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64" OR IOS_ARCH MATCHES "^arm64.*")
     SET_PROPERTY(SOURCE ${XNNPACK_AARCH64_ASM_MICROKERNEL_SRCS} APPEND_STRING PROPERTY COMPILE_FLAGS " -arch ${IOS_ARCH} ")
   ENDIF()
 ENDIF()
-IF(CMAKE_SYSTEM_PROCESSOR MATCHES "^(i[3-6]86|x86_64)$" OR IOS_ARCH MATCHES "^(i386|x86_64)$")
+IF(CMAKE_SYSTEM_PROCESSOR MATCHES "^(i[3-6]86|x86_64|AMD64)$" OR IOS_ARCH MATCHES "^(i386|x86_64|AMD64)$")
   SET_PROPERTY(SOURCE ${XNNPACK_SSE_MICROKERNEL_SRCS} APPEND_STRING PROPERTY COMPILE_FLAGS " -msse ")
   SET_PROPERTY(SOURCE ${XNNPACK_SSE2_MICROKERNEL_SRCS} APPEND_STRING PROPERTY COMPILE_FLAGS " -msse2 ")
   SET_PROPERTY(SOURCE ${XNNPACK_SSE41_MICROKERNEL_SRCS} APPEND_STRING PROPERTY COMPILE_FLAGS " -msse4.1 ")
   SET_PROPERTY(SOURCE ${XNNPACK_AVX_MICROKERNEL_SRCS} APPEND_STRING PROPERTY COMPILE_FLAGS " -mavx ")
   SET_PROPERTY(SOURCE ${XNNPACK_FMA3_MICROKERNEL_SRCS} APPEND_STRING PROPERTY COMPILE_FLAGS " -mfma ")
   SET_PROPERTY(SOURCE ${XNNPACK_AVX2_MICROKERNEL_SRCS} APPEND_STRING PROPERTY COMPILE_FLAGS " -mfma -mavx2 ")
-  SET_PROPERTY(SOURCE ${XNNPACK_AVX512F_MICROKERNEL_SRCS} APPEND_STRING PROPERTY COMPILE_FLAGS " -mavx512f ")
+  SET_PROPERTY(SOURCE ${XNNPACK_AVX512F_MICROKERNEL_SRCS} APPEND_STRING PROPERTY COMPILE_FLAGS " -mavx512f -fno-asynchronous-unwind-tables ")
```

Comment:
`-fno-asynchronous-unwind-tables` looks out-of-place here. Why is it needed?

Reply:
Because error occured.
```
C:\Users\mattn\AppData\Local\Temp\ccKOZQIe.s:44: Error: invalid register for .seh_savexmm
C:\Users\mattn\AppData\Local\Temp\ccKOZQIe.s:46: Error: invalid register for .seh_savexmm
C:\Users\mattn\AppData\Local\Temp\ccKOZQIe.s:48: Error: invalid register for .seh_savexmm
C:\Users\mattn\AppData\Local\Temp\ccKOZQIe.s:50: Error: invalid register for .seh_savexmm
C:\Users\mattn\AppData\Local\Temp\ccKOZQIe.s:52: Error: invalid register for .seh_savexmm
C:\Users\mattn\AppData\Local\Temp\ccKOZQIe.s:54: Error: invalid register for .seh_savexmm
C:\Users\mattn\AppData\Local\Temp\ccKOZQIe.s:56: Error: invalid register for .seh_savexmm
C:\Users\mattn\AppData\Local\Temp\ccKOZQIe.s:58: Error: invalid register for .seh_savexmm
C:\Users\mattn\AppData\Local\Temp\ccKOZQIe.s:60: Error: invalid register for .seh_savexmm
C:\Users\mattn\AppData\Local\Temp\ccKOZQIe.s:62: Error: invalid register for .seh_savexmm
C:\Users\mattn\AppData\Local\Temp\ccKOZQIe.s:64: Error: invalid register for .seh_savexmm
C:\Users\mattn\AppData\Local\Temp\ccKOZQIe.s:66: Error: invalid register for .seh_savexmm
C:\Users\mattn\AppData\Local\Temp\ccKOZQIe.s:68: Error: invalid register for .seh_savexmm
C:\Users\mattn\AppData\Local\Temp\ccKOZQIe.s:70: Error: invalid register for .seh_savexmm
C:\Users\mattn\AppData\Local\Temp\ccKOZQIe.s:72: Error: invalid register for .seh_savexmm
C:\Users\mattn\AppData\Local\Temp\ccKOZQIe.s:74: Error: invalid register for .seh_savexmm
```
This is related on SEH on Windows does not handle registers here. So I put this flag to disable Windows Structured Exception.

Reply:
According to https://gcc.gnu.org/bugzilla/show_bug.cgi?id=65782 it is fixed now. Does it still fail with the latest mingw?

Reply:
@mattn

Reply:
Yes. I use gcc 9.0 but it is still not fixed.

Reply:
Ok, it seems that https://gcc.gnu.org/bugzilla/show_bug.cgi?id=65782 was fixed just two weeks ago, and hasn't yet propagated to release binaries. Adding `-fno-asynchronous-unwind-tables` for MinGW is ok with me, but don't do it on all platforms, as it affects debugging experience. Please use
```cmake
SET_PROPERTY(SOURCE ${XNNPACK_AVX512F_MICROKERNEL_SRCS} APPEND_STRING PROPERTY COMPILE_FLAGS " -mavx512f ")
IF(MINGW)
  # Work-around for https://gcc.gnu.org/bugzilla/show_bug.cgi?id=65782
  SET_PROPERTY(SOURCE ${XNNPACK_AVX512F_MICROKERNEL_SRCS} APPEND_STRING PROPERTY COMPILE_FLAGS " -fno-asynchronous-unwind-tables ")
ENDIF()
```